src/http/web_socket.cr

```diff
@@ -100,52 +110,101 @@ class HTTP::WebSocket
     @ws.close(message)
   end
 
-  def run
+  struct Message
+    getter opcode
+
+    def initialize(@opcode : Opcode, @data : String | Bytes)
+    end
+
+    def text
+      @data.as(String)
+    end
+
+    def binary
+      @data.as(Bytes)
+    end
+  end
+
+  private def handle_opcode(message)
+    case message.opcode
+    when .close?
+      close(message.text) unless closed?
+      @close_reason = message.text
+    when .ping?
+      pong(message.text) unless closed?
+    end
+
+    if message.opcode.text? || message.opcode.binary?
```

Comment:
This could be simplified to just `message.opcode.text? || message.opcode.binary?`. `return` is implicit anyway.